src/core/website_manager.py

```diff
@@ -1,311 +1,218 @@
+import http
 import json
 import os
 import re
 import time
-import traceback
-from dataclasses import dataclass, field
-from json import JSONDecodeError
-from typing import NoReturn, Optional
+from dataclasses import dataclass
+from logging import Logger
+from typing import Any, NoReturn
 from urllib.parse import urlparse
 
 import requests
 from bs4 import BeautifulSoup
-from tldextract.suffix_list import SuffixListNotFound
 from tldextract.tldextract import TLDExtract
 
-from lib.constants import (
-    MESSAGE_HAR,
-    MESSAGE_HEADERS,
-    MESSAGE_HTML,
-    MESSAGE_URL,
-    SCRIPT,
-)
-from lib.logger import get_logger
+from app.communication import Message
+from lib.constants import SCRIPT
 from lib.settings import SPLASH_HEADERS, SPLASH_TIMEOUT, SPLASH_URL
 from lib.timing import global_start
 from lib.tools import get_unique_list
 
 
-@dataclass
+@dataclass  # fixme: make frozen
```

Comment:
What stops you from freezing it now?

Reply:
Currently, the website manager modifies the `values` field with the result returned from the extractors. This values field is then used with the decide method of the extractors. Once this is untangled, we can remove the value field and make it frozen.